internal/pkg/agent/cmd/upgrade_test.go

```diff
@@ -68,6 +82,245 @@ func TestUpgradeCmd(t *testing.T) {
 		// this makes sure all client assertions are done
 		<-clientCh
 	})
+	t.Run("fail if fleet managed and unprivileged", func(t *testing.T) {
+		// Set up mock TCP server for gRPC connection
+		tcpServer, err := net.Listen("tcp", "127.0.0.1:")
+		require.NoError(t, err)
+		defer tcpServer.Close()
+
+		s := grpc.NewServer()
+		defer s.Stop()
+
+		// Define mock server and agent information
+		upgradeCh := make(chan struct{})
+		mock := &mockServer{upgradeStop: upgradeCh}
+		mockAgentInfo := mockinfo.NewAgent(t)
+		mockAgentInfo.On("IsStandalone").Return(false) // Simulate fleet-managed agent
+		mockAgentInfo.On("Unprivileged").Return(true)  // Simulate unprivileged mode
```

Comment:
Why not setting expectations on the mock using something like:
```suggestion
		mockAgentInfo.EXPECT().IsStandalone().Return(false) // Simulate fleet-managed agent
		mockAgentInfo.EXPECT().Unprivileged().Return(true)  // Simulate unprivileged mode
```